src/EmojiList.js

```diff
@@ -1,19 +1,28 @@
 import React, { Component } from 'react';
 import { connect } from 'react-redux';
 import ItemComponent from './EmojiComponent';
+import SettingsModeButtons from './SettingsModeButtons';
 
 const apply = connect(
-    state => ({ smiles: state.items.smiles })
+    state => ({
+        smiles: state.items.smiles,
```

Comment:
This is best practice and common advice to avoid selecting something in your state directly. It is better to create selectors. E.g. in `redux` folder you can create file `selectors.js` and fill it with following code:
```javascript
const selectSmiles = state => state.items.smiles;
const selectMode = state => state.pages.mode;
```
and then replace this code with following:
```javascript
state => ({
  smiles: selectSmiles(state),
  mode: selectMode(mode),
})
```

Reply:
fixed